### PR TITLE
Change credential timeout.

### LINKED ISSUE #259: CLI took > 30 seconds to tell me that it cannot access to S3
```
cargo r new --index-uri s3://quickwit-dev/my-hdfs --index-config-path ./hdfslogs_index_config.json --overwrite
Finished dev [unoptimized + debuginfo] target(s) in 0.37s 
Running `target/debug/quickwit new --index-uri 's3://quickwit-dev/my-hdfs' --index-config-path ./hdfslogs_index_config.json --overwrite`
Command failed: Another error occured. `Metastore error`. Cause: `StorageError(kind=Unauthorized, source=Failed to fetch object: s3://quickwit-dev/my-hdfs/quickwit.json)
```

### PR DESCRIPTION
**Pull request by mosuka**
### Context / purpose
https://github.com/quickwit-inc/quickwit/issues/259

### Description
Change credential timeout to 5secs from the default (30secs).

### How was this PR tested?
Rename the credential file to a different name, and then run the following command:
```
cargo r new --index-uri s3://quickwit-dev/my-hdfs --index-config-path ./hdfslogs_index_config.json --overwrite
```

### Checklist
*Remove or complete this list if required.*
- [x] tested locally

Closes #259

### CODE REVIEW

File: quickwit-storage/src/object_storage/s3_compatible_storage.rs [hunk 1]
```diff
@@ -71,7 +74,9 @@ impl fmt::Debug for S3CompatibleObjectStorage {
 }
 
 fn create_s3_client(region: Region) -> anyhow::Result<S3Client> {
-    let credentials_provider = DefaultCredentialsProvider::new()
+    let mut chain_provider = ChainProvider::new();
+    chain_provider.set_timeout(Duration::from_secs(5));
```
Comment (fmassot) [inline quickwit-storage/src/object_storage/s3_compatible_storage.rs:78]: can you put this in a const variable?

Comment (mosuka) [reply quickwit-storage/src/object_storage/s3_compatible_storage.rs]: Added const variables.
Thanks.

File: quickwit-storage/src/object_storage/s3_compatible_storage.rs [hunk 2]
```diff
@@ -55,6 +55,12 @@ use crate::retry::{retry, IsRetryable, Retry};
 use crate::{PutPayload, Storage, StorageErrorKind};
 use crate::{StorageError, StorageResult};
 
+/// A credential timeout.
+const CREDENTIAL_TIMEOUT: u64 = 5;
+
+/// An timeout for idle sockets being kept-alive.
+const POOL_IDIE_TIMEOUT: u64 = 10;
```
Comment (fmassot) [inline quickwit-storage/src/object_storage/s3_compatible_storage.rs:62]: typo in variable name `POOL_IDLE_TIMEOUT`

Comment (mosuka) [reply quickwit-storage/src/object_storage/s3_compatible_storage.rs]: Oh, sorry. I made a typo...
I fixed it.

**Review by fmassot — APPROVED** [review]

**Pull request merged**